compiler/rustc_resolve/src/effective_visibilities.rs

```diff
@@ -163,6 +203,22 @@ impl<'r, 'a, 'tcx> EffectiveVisibilitiesVisitor<'r, 'a, 'tcx> {
         }
     }
 
+    fn ambiguous_effective_vis_or_private(
+        &mut self,
+        parent_id: ParentId<'a>,
+    ) -> EffectiveVisibility {
+        // Private nodes are only added to the table for caching, they could be added or removed at
+        // any moment without consequences, so we don't set `changed` to true when adding them.
+        *match parent_id {
+            ParentId::Def(def_id) => self
+                .ambiguous_def_effective_visibilities
+                .effective_vis_or_private(def_id, || self.r.private_vis_def(def_id)),
+            ParentId::Import(binding) => self
+                .ambiguous_import_effective_visibilities
+                .effective_vis_or_private(binding, || self.r.private_vis_import(binding)),
```

Comment:
This function is certainly not correct.
*Parent* of an ambiguous import is not necessarily an ambiguous import, in that case it is not in this table (it is in the regular non-ambiguous table) and the whole logic breaks.

Reply:
If the parent is ambiguous it should be looked up in the ambiguous tables, and if it's not ambiguous then it should be looked up in the regular tables.

Reply:
TBH, I really dislike where this is going, it's probably better to revert to keeping everything in the regular tables.
I'll look in more detail today/tomorrow.

Reply:
> TBH, I really dislike where this is going, it's probably better to revert to keeping everything in the regular tables. I'll look in more detail today/tomorrow.

I reverted to previous commit using regular table. Will address some of the review comments tomorrow.